### PR TITLE
Scatter Plots

### PR DESCRIPTION
**Pull request by tpreusse**
New chart type: scatter plots! Make a scientist out of every reader.
With nice tooltips for endless research 🕵️‍♀️

[screen shot 2018-10-12 at 13 27 08](https://r-styleguide-pr-177.herokuapp.com/charts/scatterplots)

_Go forth and make correlations visible, and tell everyone it proofs nothing!_

- [new scatter plot doc page](https://r-styleguide-pr-177.herokuapp.com/charts/scatterplots).

Other changes
- replace decimal point with decimal comma as discussed with Daniel Meyer
- add zero width reset `tspan` after sub/sup `tspan` in svg text

### CODE REVIEW

File: src/components/Chart/ContextBox.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+import React from 'react'
+import {css} from 'glamor'
+import PropTypes from 'prop-types'
+
+import colors from '../../theme/colors'
+import { Label } from '../Typography'
+import { sansSerifRegular14 } from '../Typography/styles'
+
+const boxStyle = css({
```
Comment (garamond) [inline src/components/Chart/ContextBox.js]: Diverges a bit from the visual design of Republik:

- `boxShadow` is too fuzzy
- `borderRadius` is unusual

File: src/components/Chart/ContextBox.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    right: css({bottom: -8, transform: 'translateX(50%)', right: '15%'})
+  },
+  below: {
+    center: css({top: -8, transform: 'translateX(-50%) rotate(180deg)', left: '50%'}),
+    left: css({top: -8, transform: 'translateX(-50%) rotate(180deg)', left: '15%'}),
+    right: css({top: -8, transform: 'translateX(50%) rotate(180deg)', right: '15%'})
+  }
+}
+
+const labledValueStyle = css({
```
Comment (garamond) [inline src/components/Chart/ContextBox.js]: Typo

File: src/components/Chart/ContextBox.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  }
+})
+
+export const ContextBoxValue = ({label, children}) => {
+  if (!children) {
+    return null
+  }
+  return (
+    <div {...labledValueStyle}>
+      {!!label && <Label>{label}<br /></Label>}
```
Comment (garamond) [inline src/components/Chart/ContextBox.js]: I think the the label should have a more distinctive style in order to be distinguishable from the children.

File: src/components/Chart/ScatterPlots.js [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+          {this.state.hover.map((symbol, i) => (
+            <circle key={`hover${symbol.key}`}
+              fill='none'
+              stroke='#000'
+              cx={symbol.cx}
+              cy={symbol.cy}
+              r={symbol.r} />
+          ))}
+          {
+            yTicks.map((tick, i) => (
```
Comment (garamond) [inline src/components/Chart/ScatterPlots.js]: Maybe the rendering of the axes could be encapsulated in separate methods – along with the axis/ticks calculations above. Haven't tried it, though.

File: src/components/Chart/ScatterPlots.js [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+    if (top === undefined || left === undefined || !this.symbols) {
+      return
+    }
+
+    let hoverTouch = false
+    let currentEvent = event
+    if (currentEvent.nativeEvent) {
+      currentEvent = event.nativeEvent
+    }
+    while (currentEvent.sourceEvent) {
```
[the remark this answers is not in the document]
Comment (tpreusse) [reply src/components/Chart/ScatterPlots.js]: Good catch. I copied that blindly from d3, it's a d3 internal property we do not need to support. I'll remove.

**Comment by tpreusse** [comment]
Thank you for your very valuable feedback @garamond.

I improve the context box design with your inputs:
screen shot 2018-10-13 at 15 00 40
- no more border radius
- a strong label
- the less fuzzy box shadow from the dropdown

I also did the code cleanups and naming improvements. And opened issues for the [touch interaction](https://github.com/orbiting/styleguide/issues/179) and [axis refactoring](https://github.com/orbiting/styleguide/issues/178) – a bit harder and less clear task I'd like to move to the next weeks.

**Pull request merged**